examples/gradient_descent.py

```diff
@@ -0,0 +1,85 @@
+"""
+Gradient descent on a sphere.
+"""
+
+import numpy as np
+import matplotlib
+matplotlib.use("Agg")  # NOQA
+import matplotlib.pyplot as plt
+import matplotlib.animation as manimation
+
+from geomstats.hypersphere import Hypersphere
+import geomstats.visualization as visualization
+import geomstats.vectorization as vectorization
+
+SPHERE2 = Hypersphere(dimension=2)
+METRIC = SPHERE2.metric
+
+
+def gradient_descent(start,
+                     loss,
+                     grad,
+                     manifold,
+                     lr=0.05,
+                     max_iter=128,
+                     precision=1e-5):
+    x = start
+    for i in range(max_iter):
+        x_prev = x
+        euclidian = - lr * grad(x)
```

Comment:
euclidean*. why not euclidean_grad?

Reply:
originally to limit space - but was waiting for your comment to check if it was super horrible to read :)